ui/src/spinbox.tsx

```diff
@@ -108,16 +111,23 @@ export const
       }, [max, min, parseValue, precision]),
       handleOnInput = React.useCallback((val: U) => {
         wave.args[name] = val
+        m.value = val
         if (trigger) wave.push()
+        // eslint-disable-next-line react-hooks/exhaustive-deps
       }, [name, trigger]),
       debouncedHandleOnInput = React.useRef(wave.debounce(DEBOUNCE_TIMEOUT, handleOnInput)),
       onInput = (e: React.SyntheticEvent<HTMLElement>) => {
         const numVal = handleValue((e.target as HTMLInputElement).value)
         trigger ? debouncedHandleOnInput.current(numVal) : handleOnInput(numVal)
       }
 
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    React.useEffect(() => { wave.args[name] = (value < min) ? min : ((value > max) ? max : value) }, [])
+    React.useEffect(() => {
+      if (value === wave.args[name]) return
+      const val = (value < min) ? min : ((value > max) ? max : value)
+      wave.args[name] = val
+      setVal(String(val))
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [value])
```

Comment:
We probably want to include all the necessary dependencies than just `value`, right? This way the `value` would be always in bounds of minmax.

If so, please add a quick unit test validating the value is changed properly on minmax update. Should be pretty fast.

Reply:
Do you mean if the value is updated through wave app, it can also be outside of the min max boundaries?

Reply:
If the min/max is updated and the value is suddenly out of bounds, it should be adjusted, right? Also the above suffers from stale closure problem as min/max can be stale within useEffect during value change.

Or is there any reason for including just `value`  as the only dependency?

Reply:
Makes sense, thanks for the explanation. No, there is no specific reason.